configen/FileGenerator.swift

```diff
@@ -94,35 +94,72 @@ struct FileGenerator {
     }
     
     var line = ""
+    var finalType = type
+    var isObjcOptional = false
+    
+    if optionsParser.isObjC && type.contains("?") {
+      finalType = type.replacingOccurrences(of: "?", with: "")
+      isObjcOptional = value as? String == "nil" ? true : false
+    }
     
-    switch (type) {
+    switch (finalType) {
     case ("Double"):
-      line = template.doubleImplementation
+      if isObjcOptional && template is ObjectiveCTemplate{
+        let objcTemplate = template as! ObjectiveCTemplate
+        line = objcTemplate .optionalImplementation
+        line.replace(token: objcTemplate.declarationToken, withString: objcTemplate.doubleDeclaration)
```

Comment:
Actually, it might be nice to pull these three lines out into a function as they are repeated quite a lot below. Maybe have something like `optionalObjectiveCLine(for template: HeaderTemplate, declaration: String) -> String`. Only a suggestion, see what you think